File: swt/src/ext.r/bugfm.f
      INTEGER LDATE(32),TIME(9),USERID(33),UNAME(102),PNAME(102),ENAME(1
     *02)
      INTEGER INPUT
      IF((INPUT(-10,'Your name: *,,`s.',UNAME).EQ.-1))GOTO 10001
      IF((INPUT(-10,'Name of problem software: *,,`s.',PNAME).EQ.-1))GOT
     *O 10001
      IF((INPUT(-10,'Name of example file (<CR> if none): *s.',ENAME).EQ
     *.-1))GOTO 10001
      GOTO 10000
10001   CALL SETERR(1000)
        CALL SWT
10000 CALL DATE(7,LDATE)
      CALL DATE(2,TIME)
      CALL DATE(3,USERID)
      CALL PRINT(-11,'*s at *s*2n.',LDATE,TIME)
      CALL PRINT(-11,'By: *s (*s)*n.',UNAME,USERID)
      CALL PRINT(-11,'Re: *s*n.',PNAME)
      IF((ENAME(1).EQ.0))GOTO 10002
        CALL PRINT(-11,'Ex: *s*n.',ENAME)
10002 CALL REMARK('Description of bug (end with ctrl\c in column 1).')
      CALL PRINT(-11,'*nDescription:*2n.')
      CALL FCOPY(-10,-11)
      CALL SWT
      END
C ---- Long Name Map ----

File: swt/src/ext.r/bugfm.r
# bugfm --- generate a bug report

   character ldate (32), time (9), userid (MAXUSERNAME), uname (MAXLINE),
         pname (MAXLINE), ename (MAXLINE)

   integer input

   if (input (STDIN, "Your name: *,,`s"p, uname) == EOF
         ||  input (STDIN, "Name of problem software: *,,`s"p,
                  pname) == EOF
         || input (STDIN, "Name of example file (<CR> if none): *s"p,
                  ename) == EOF) {
      call seterr (1000)   # make our caller abort
      stop
      }

   call date (SYS_LDATE, ldate)
   call date (SYS_TIME, time)
   call date (SYS_USERID, userid)

   call print (STDOUT, "*s at *s*2n"p, ldate, time)
   call print (STDOUT, "By: *s (*s)*n"p, uname, userid)
   call print (STDOUT, "Re: *s*n"p, pname)
   if (ename (1) ~= EOS)
      call print (STDOUT, "Ex: *s*n"p, ename)

   call remark ("Description of bug (end with ctrl\c in column 1)"p)
   call print (STDOUT, "*nDescription:*2n"p)
   call fcopy (STDIN, STDOUT)

   stop
   end
